src/observers/OmnisharpStatusBarObserver.ts

```diff
@@ -0,0 +1,191 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as serverUtils from '../omnisharp/utils';
+import { CompositeDisposable, Subject, Scheduler } from 'rx';
+import { DocumentFilter, DocumentSelector, StatusBarItem, vscode } from '../vscodeAdapter';
+import { OmniSharpServer } from '../omnisharp/server';
+import { Status } from './status';
+import { basename } from 'path';
+import { OmnisharpServerOnServerError, BaseEvent, OmnisharpOnMultipleLaunchTargets, OmnisharpOnBeforeServerInstall, OmnisharpOnBeforeServerStart, ActiveTextEditorChanged, OmnisharpServerOnStop, OmnisharpServerOnStart } from "../omnisharp/loggingEvents";
+
+let defaultSelector: DocumentSelector = [
+    'csharp', // c#-files OR
+    { pattern: '**/project.json' }, // project.json-files OR
+    { pattern: '**/*.sln' }, // any solution file OR
+    { pattern: '**/*.csproj' }, // an csproj file
+    { pattern: '**/*.csx' }, // C# script
+    { pattern: '**/*.cake' } // Cake script
+];
+
+export class OmnisharpStatusBarObserver {
+    private defaultStatus: Status;
+    private projectStatus: Status;
+    private localDisposables: CompositeDisposable;
+    private updateProjectDebouncer: Subject<OmniSharpServer>;
+    private firstUpdateProject: boolean;
+
+    constructor(private vscode: vscode, private statusBar: StatusBarItem, scheduler?: Scheduler) {
+        this.defaultStatus = new Status(defaultSelector);
+        this.updateProjectDebouncer = new Subject<OmniSharpServer>();
+        this.updateProjectDebouncer.debounce(1500, scheduler).subscribe((server: OmniSharpServer) => { this.updateProjectInfo(server); });
```

Comment:
If we're using debouce from RX, can we remove the dependency on `lodash.debounce` in package.json?

Reply:
Done!

Reply:
thanks!